processing/src/main/java/org/apache/druid/segment/loading/DataSegmentKiller.java

```diff
@@ -54,6 +55,29 @@ static String descriptorPath(String path)
    */
   void kill(DataSegment segment) throws SegmentLoadingException;
 
+  /**
+   * Kills a list of segments from deep storage. The default implementation calls kill on the segments in a loop.
+   * Implementers of this interface can leverage batch / bulk deletes to be more efficient. It is preferable to attempt
+   * to delete all segments even if there is an issue with deleting a single one. This is up to implementers to
+   * implement as putting a try catch around the default kill via iteration can be problematic if the client of the deep
+   * storage is unable to authenticate itself and segment loading exception doesn't encode enough information in it to \
+   * understand why it failed.
+   * <p>
+   * If a segment or segments does not exist in deep storage, the method should not throw an exception.
+   * <p>
+   * This version of kill must **NOT** require additional permissions on the deep storage beyond what
+   * {@link #kill(DataSegment)} requires.
+   * @param segments The list of segments to kill.
+   * @throws SegmentLoadingException If there is an exception during deletion such as a segment in the list could not be
+   * completely removed.
+   */
+  default void kill(List<DataSegment> segments) throws SegmentLoadingException
```

Comment:
We could consider extending this to `killQuietly`, as that is called from the appenderator layer for multiple segments at a time.

Reply:
for example [here](https://github.com/apache/druid/blob/28914bbab82480f33a345913a30c32321d36fc37/server/src/main/java/org/apache/druid/segment/realtime/appenderator/BaseAppenderatorDriver.java#L505)

Reply:
I agree. This was chosen not to be done in this PR as it would require more testing